### ISSUE TITLE
Fix S6964 FP: Don't raise on properties annotated with the BindRequiredAttribute

### ISSUE THREAD

**Issue by PaulBol** [SonarSource/sonar-dotnet#9690]
### Description

[S6964](https://sonarsource.github.io/rspec/#/rspec/S6964/csharp) complains about under-posting when properties are not nullable and annotated with [`BindRequiredAttribute`](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.modelbinding.bindrequiredattribute). 

> Indicates that a property is required for model binding. When applied to a property, the model binding system requires a value for that property.

### Repro steps

Use view model with property

```cs
[BindRequired] public int PasswordMinLength { get; init; }
```

### Expected behavior

No issue reported.

### Actual behavior

[S6964](https://sonarsource.github.io/rspec/#/rspec/S6964/csharp) issue reported.

### Known workarounds

Use `required` keyword in addition. Needs further code changes.

### Related information

* C#/VB.NET Plugins version: n/a
* Visual Studio version: 2022
* MSBuild / dotnet version: .NET 6
* SonarScanner for .NET version (if used): 5.14 (.NET Framework)
* Operating System: Windows 10

**Comment by sebastien-marichal** [comment]
Hello @PaulBol,

Thank you for reporting this issue to us.

I confirm this is a false positive!

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1537](https://sonarsource.atlassian.net/browse/NET-1537)

[NET-1537]: https://sonarsource.atlassian.net/browse/NET-1537?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by martin-strecker-sonarsource** [comment]
Fixed in version [10.15.0.120848](https://github.com/SonarSource/sonar-dotnet/releases/tag/10.15.0.120848)
